src/content/docs/building-with-kessel/how-to/migrate-from-rbac-v1-to-v2.mdx

```diff
@@ -4,4 +4,223 @@ sidebar:
+
+The workspace ids are obtained using the [StreamedListObjects](https://buf.build/project-kessel/inventory-api/docs/main:kessel.inventory.v1beta2#kessel.inventory.v1beta2.KesselInventoryService.StreamedListObjects) method and its parameters should be defined as follows:
+
+```go
+objectType := &kesselv2.RepresentationType{
+	ResourceType: "workspace",
+	ReporterType: "rbac",
+},
+
+relation = // the permission to check for
+
+subject := &kesselv2.SubjectReference{
+	Resource: &kesselv2.ResourceReference{
+		ResourceType: "principal",
+		ResourceId:   principalId,
+		Reporter: &kesselv2.ReporterReference{
+			Type: "rbac",
+		},
+	},
+}
+```
+
+<Aside type="note" title="Supporting all principal types">
+Note that the principal id will differ based on the type of the principal:
+
+* user principal \- `identity.User.Username`
+* service account \- `identity.ServiceAccount.Username`
+</Aside>
+
+Native, workspace-level list pattern implementations and PoCs:
+* [Digital Roadmap](https://github.com/RedHatInsights/digital-roadmap-backend/pull/182)
+* [Resource Optimization Service](https://github.com/josejulio/ros-backend/pull/1/files)
+
+See [Native, workspace-level list](https://docs.google.com/document/d/1XnINsHuYeHEi22q_1cS0gUalX-eXl3V19gGf0Wr8NsE/edit?tab=t.0#heading=h.17ytnrwi9six) for additional information.
+
+### Adding tests
+
+TBD
+
+[Example of a testcase for Kessel integration](https://github.com/RedHatInsights/config-manager/blob/a937c5f283fb3acb2931cf2713173f220cd1bd11/internal/http/middleware/authorization/kessel_test.go)
+
+### Add configuration options for the Kessel client
+
+The application should add the following configuration options for Kessel:
+
+* `KESSEL_ENABLED`
+* `KESSEL_URL`
+* `KESSEL_AUTH_ENABLED`
+* `KESSEL_AUTH_CLIENT_ID`
+* `KESSEL_AUTH_CLIENT_SECRET`
+* `KESSEL_AUTH_OIDC_ISSUER`
+* `KESSEL_INSECURE`
+
+The ClowdApp template should also be updated accordingly to enable environment-specific configuration.
+See [Config Manager’s configuration](https://github.com/RedHatInsights/config-manager/pull/242/commits/a937c5f283fb3acb2931cf2713173f220cd1bd11#diff-55823110ec960277fef2e07ee7d7c8f310baff0bd789e5b27160a1ccb6b4a81e) for an example.
+
+## Validate the changes
+
+* see [Ephemeral Deployments with Kessel Inventory API](https://cuddly-tribble-gq7r66v.pages.github.io/using-kessel/inventory-api/ephemeral/) for information on how to deploy Kessel in the ephemeral environment.
+* [insights-service-deployer] is a feature-rich script for end-to-end deployment and configuration of Kessel related services in the ephemeral environment.
+
+## Deploy the changes
+
+Follow [Using Kessel in production](https://github.com/project-kessel/docs-internal/blob/a7e66c32d491880958c399584725267efbb678f6/src/content/docs/using-kessel/in-production.mdx) to set up your service to use Kessel in stage and prod.
```

Comment:
For internal links, we can make them nicer if we use https://project-kessel.github.io/docs-internal/

e.g. https://project-kessel.github.io/docs-internal/using-kessel/in-production

(once it's merged)

Reply:
Fixed